src/core/tsi/ssl_transport_security.cc

```diff
@@ -953,7 +953,10 @@ static void ssl_protector_destroy(tsi_frame_protector* self) {
   tsi_ssl_frame_protector* impl =
       reinterpret_cast<tsi_ssl_frame_protector*>(self);
   if (impl->buffer != nullptr) gpr_free(impl->buffer);
-  if (impl->ssl != nullptr) SSL_free(impl->ssl);
+  if (impl->ssl != nullptr) {
+    SSL_shutdown(impl->ssl);
```

Comment:
Don't we need to add the shutdown API at `ssl_handshaker_result_destroy` and `ssl_handshaker_destroy`? Also, since we are calling `SSL_shutdown` explicitly now, do we still need to call `SSL_CTX_set_quiet_shutdown`?